src/backend/dist.js

```diff
@@ -67,42 +68,66 @@ const downloadAudioFile = function(url, filePath, next) {
 
 };
 
-const downloadJson = function(appPath, endpoint, jsonFile, cb) {
+const downloadJsonFromGithub = function(appPath, endpoint, jsonFile, cb) {
   const fileStream = fs.createWriteStream(appPath + '/json/' + jsonFile);
+  var statusCode;
 
   fileStream.on('error', function(err) {
     console.log(err);
+    return cb(err);
   });
 
+  fileStream.on('finish', function () {
+    if(statusCode != 200) {
+      return cb(new Error('Response = ' + statusCode + ' received'));
+    }
+    return cb();
+  });
 
   try {
-    console.log('Downloading ' + endpoint + '/' + jsonFile);
+    console.log('Downloading ' + endpoint);
     request
-      .get({url:endpoint + '/' + jsonFile,timeout:30000})
+      .get({url:endpoint, timeout:30000})
```

Comment:
on what parameters, this value is set to 30000

Reply:
Nothing much. Just thought that this way a decent amount of time for the request module to wait for a response from a link before timing out and throwing an error that the server is not responding and the user is most probably offline.